core/parsigex/parsigex.go

```diff
@@ -158,72 +159,100 @@ func NewEth2Verifier(eth2Cl eth2wrap.Client, pubSharesByKey map[core.PubKey]map[
 			return errors.New("invalid shareIdx")
 		}
 
+		epoch, err := epochFromSlot(ctx, eth2Cl, eth2p0.Slot(duty.Slot))
+		if err != nil {
+			return err
+		}
+
+		sigRoot, err := data.SignedData.MessageRoot()
+		if err != nil {
+			return err
+		}
+
 		switch duty.Type {
 		case core.DutyAttester:
-			att, ok := data.SignedData.(core.Attestation)
-			if !ok {
-				return errors.New("invalid attestation")
+			err = signing.VerifySignedData(ctx, eth2Cl, signing.DomainBeaconAttester, epoch, sigRoot,
+				data.Signature().ToETH2(), pubshare)
+			if err != nil {
+				return errors.Wrap(err, "invalid attestation", z.Str("duty", duty.String()))
 			}
 
-			return signing.VerifyAttestation(ctx, eth2Cl, pubshare, &att.Attestation)
+			return nil
 		case core.DutyProposer:
-			block, ok := data.SignedData.(core.VersionedSignedBeaconBlock)
-			if !ok {
-				return errors.New("invalid block")
+			err = signing.VerifySignedData(ctx, eth2Cl, signing.DomainBeaconProposer, epoch, sigRoot,
```

Comment:
what about adding `Domain()` to `SignedData` interface?

Reply:
Can then probably add ` core.VerifySignature(context.Context, eth2wrap.Client, slot, core.SignedData, eth2p0.Pubkey)`

Reply:
Or can skip adding domain to signeddata and do `core.VerifySignature(context.Context, eth2wrap.Client, core.Duty, core.SignedData, eth2p0.Pubkey)` and infer domain from duty type.

Reply:
yeah we can map duty type to corresponding domain

Reply:
or maybe add a `Domain()` method to `core.Duty` type?

Reply:
less coupling is better, we can always couple more later if required